packages/@aws-cdk/aws-lambda-python/lib/Dockerfile

```diff
@@ -3,12 +3,16 @@
 ARG IMAGE=public.ecr.aws/sam/build-python3.7
 FROM $IMAGE
 
+# Upgrade pip (required by cryptography v3.4 and above, which is a dependency of poetry)
+RUN pip install --upgrade pip
+RUN pip install pipenv poetry
```

Comment:
Moved this up since if using Code Artifact, this install uses the private repo, and causes the cache to not be used, forcing rebuilds of the Docker image every time.